py_mmd_tools/mmd_to_nc.py

```diff
@@ -0,0 +1,256 @@
+        try:
+            out['keywords_vocabulary'] = ':'.join([prefix, element.find(
+                'mmd:resource', namespaces=self.namespaces).text])
+            sep['keywords_vocabulary'] = self.mmd_yaml['keywords']['vocabulary']['separator']
+        except AttributeError:
+            print('No resource for keyword')
+
+        return out, sep
+
+    def get_title_abstract(self, element, tag):
+        """
+        """
+
+        out = {}
+        acdd_name, sep = self.get_acdd(self.mmd_yaml[tag][tag])
+        # Check element language
+        for att in element.attrib:
+            if element.attrib[att] == 'no':
+                acdd_name = acdd_name + '_no'
+            elif element.attrib[att] == 'en':
+                out[acdd_name + '_lang'] = 'en'
+            else:
+                print(f'Case not implemented for language {element.attrib[att]}')
+                return None
+        out[acdd_name] = element.text
+        return out
+
+    def get_metadata_identifier(self, element):
+        """
+        """
+
+        out = {}
+        value = element.text
+        try:
+            out['id'] = value.split(':')[1]
+            out['naming_authority'] = value.split(':')[0]
+        except IndexError:
+            out['id'] = value
+
+        return out
+
+    def update_nc(self):
+        """
+        Update a netcdf file global attributes.
+        """
+
+        acdd = {}
+
+        # Loop on expected MMD elements
+        # ie from mmd_elements.yaml = kind-off mapping between acdd and mmd
+        for mmd_element in self.mmd_yaml:
+
+            elements = self.tree.findall('mmd:' + mmd_element, self.namespaces)
+
+            if len(elements) == 0:
+                print(f'{mmd_element} not found in input MMD file')
+                continue
+
+            # Loop on  MMD elements found
+            for elem in elements:
+
+                tag = ET.QName(elem).localname
+
+                if mmd_element in ['title', 'abstract']:
+                    match = self.get_title_abstract(elem, tag)
+                    acdd = self.update_acdd(acdd, match)
+
+                elif mmd_element == 'keywords':
+                    match, sep = self.get_keyword(elem)
+                    acdd = self.update_acdd(acdd, match, sep)
+
+                elif mmd_element == 'last_metadata_update':
+                    match, sep = self.get_last_metadata_update(elem)
+                    acdd = self.update_acdd(acdd, match, sep)
+
+                elif mmd_element == 'metadata_identifier':
+                    match = self.get_metadata_identifier(elem)
+                    acdd = self.update_acdd(acdd, match)
+
+                elif mmd_element == 'personnel':
+                    match, sep = self.get_personnel(elem)
+                    acdd = self.update_acdd(acdd, match, sep)
+
+                # dataset_citation = repeat of mmd fields already processed elsewhere
+                elif mmd_element == 'dataset_citation':
```

Comment:
I'm not 100% sure what to do here, as some elements from the mmd are matched several times to acdd elements.

Reply:
I'm not sure if I understand your question. Citation doesn't seem to be an element in ACDD, so it can be left out. This is the advantage of writing the MMD first. Then we only need to write the most important metadata in the netcdf file, without having an ambiguous MMD.